app/controllers/document_lead_image_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class DocumentLeadImageController < ApplicationController
+  def index
+    @document = Document.find_by_param(params[:document_id])
+  end
+
+  def create
+    document = Document.find_by_param(params[:document_id])
+    upload = UploadedImageService.new(params.require(:image)).process
+
+    if upload.valid?
+      image = create_image_from_upload(upload, document)
+      redirect_to edit_document_lead_image_path(params[:document_id], image.id)
+    else
+      redirect_to document_lead_image_path, alert: upload.errors
+    end
+  end
+
+  def edit
+    @document = Document.find_by_param(params[:document_id])
+    @image = Image.find_by(id: params[:image_id])
+  end
+
+  def update
+    document = Document.find_by_param(params[:document_id])
+    image = Image.find_by(id: params[:image_id])
+    image.update(update_params)
+    document.update(lead_image_id: image.id)
+    if params[:next] == "lead-image"
+      return redirect_to document_lead_image_path(document)
```

Comment:
Little one: we're not actually returning the redirect, so it should really be on a separate line.